services/subscriber/service.go

```diff
@@ -183,20 +183,28 @@ func (s *Service) createSubscription(se subEntry, mode string, destinations []st
 	default:
 		return nil, fmt.Errorf("unknown balance mode %q", mode)
 	}
-	writers := make([]PointsWriter, len(destinations))
-	stats := make([]writerStats, len(writers))
-	for i, dest := range destinations {
+	writers := []PointsWriter{}
```

Comment:
You can pre-allocate and still use `append`. You just need to allocate the capacity and a length of zero. For example: `writers := make([]PointsWriter, 0, len(destinations))`